src/daft-logical-plan/src/optimization/rules/push_down_filter.rs

```diff
@@ -1026,4 +1085,74 @@ mod tests {
 
         Ok(())
     }
+
+    #[rstest]
+    #[case(JoinType::Left, resolved_col("b"), JoinType::Inner)]
+    #[case(JoinType::Left, resolved_col("a"), JoinType::Left)]
+    #[case(JoinType::Right, resolved_col("a"), JoinType::Inner)]
+    #[case(JoinType::Right, resolved_col("b"), JoinType::Right)]
+    #[case(JoinType::Outer, resolved_col("a").eq(resolved_col("b")), JoinType::Inner)]
```

Comment:
Would be good to do an `eq_null_safe` too with different join types.

Reply:
changed the last one to an eq_null_safe